runelite-client/src/main/java/net/runelite/client/plugins/worldmap/mappoints/FairyRingPoint.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2018, Morgan Lewis <https://github.com/MESLewis>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.worldmap.mappoints;
+
+import com.google.common.collect.ImmutableSet;
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.util.Set;
+import javax.imageio.ImageIO;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.plugins.worldmap.WorldMapPlugin;
+import net.runelite.client.ui.overlay.worldmap.WorldMapPoint;
+
+public class FairyRingPoint extends IconWorldMapPoint
+{
+	private final static BufferedImage FAIRY_RING_ICON;
+
+	static
+	{
+			//A size of 17 gives us a buffer when triggering tooltips
+			final int ICON_BUFFER_SIZE = 17;
+			FAIRY_RING_ICON = new BufferedImage(ICON_BUFFER_SIZE, ICON_BUFFER_SIZE, BufferedImage.TYPE_INT_ARGB);
+	}
+
+	public static void setCustomIcon(boolean customIcon)
```

Comment:
This is some weird mutation of almost constant

Reply:
I couldn't find another way to get the required functionality. I'm open to suggestions.